### PR TITLE
HWP-318: Post/Community tagging api

### PR DESCRIPTION
**Pull request by BradyMitch**
# Description

This PR includes the following proposed change(s):

- Community creator (eventually moderators as well) can set tags for the community. Ex: Informative
- Community creator can remove tags from the community. This also removes all tag references on posts within the community.
- User can view community tags (that were set by community creator) as well as a count for how many times that tag was used on posts within the community.
- User can tag a post (limit 1 tag per user per post, only allows tags that were set within the community).
- User can untag a post.
- Added optional query parameter to /api/post/community/{title} to return all posts within community tagged by 'tag'. 

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested all new endpoints in swagger docs.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [x] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/server/routes/v1/post.js [hunk 1]
```diff
@@ -269,7 +289,7 @@ router.get("/community/:title", async (req, res) => {
  *        '401':
  *          description: Not Authorized. Only creator of post can edit post.
  *        '403':
- *          description: Forbidden. Community can't have more than 3 pinned posts. OR Forbidden. Can't edit creator of a post.
+ *          description: Forbidden. Community can't have more than 3 pinned posts. OR Forbidden. Can't edit tags. Use /api/post/{id}/tags instead. OR Forbidden. Can't edit creator of a post.
```
Comment (akroon3r) [inline app/server/routes/v1/post.js:292]: This description seems a little weird to read imo, lets refactor?

Comment (BradyMitch) [reply app/server/routes/v1/post.js:292]: Swagger doesn't let me use multiple descriptions or responses so I need to include them all in the same description. Do you have any suggestions for how I should write the description?

Comment (BradyMitch) [reply app/server/routes/v1/post.js:292]: This is a feature that many people on the internet seem to want but Swagger will not implement it.

Comment (akroon3r) [reply app/server/routes/v1/post.js:292]: Kk

**Pull request merged**